lib/_tls_common.js

```diff
@@ -90,80 +100,121 @@ function validateKeyOrCertOption(name, value) {
+    clientCertEngine,
+    crl,
+    dhparam,
+    ecdhCurve = tls.DEFAULT_ECDH_CURVE,
+    honorCipherOrder,
+    key,
+    minVersion,
+    maxVersion,
+    passphrase,
+    pfx,
+    privateKeyIdentifier,
+    privateKeyEngine,
+    secureProtocol,
+    sessionIdContext,
+    sessionTimeout,
+    sigalgs,
+    singleUse,
+    ticketKeys,
+  } = options;
+
+  let { secureOptions } = options;
+
+  if (honorCipherOrder)
     secureOptions |= SSL_OP_CIPHER_SERVER_PREFERENCE;
 
-  const c = new SecureContext(options.secureProtocol, secureOptions,
-                              options.minVersion, options.maxVersion);
+  const c = new SecureContext(secureProtocol, secureOptions,
+                              minVersion, maxVersion);
 
   // Add CA before the cert to be able to load cert's issuer in C++ code.
-  const { ca } = options;
+  // NOTE(@jasnell): ca, cert, and key are permitted to be falsy, so do not
+  // change the checks to !== undefined checks.
   if (ca) {
-    if (ArrayIsArray(ca)) {
-      for (const val of ca) {
-        validateKeyOrCertOption('ca', val);
-        c.context.addCACert(val);
-      }
-    } else {
-      validateKeyOrCertOption('ca', ca);
-      c.context.addCACert(ca);
-    }
+    if (ArrayIsArray(ca))
+      addCACerts(c.context, ...ca);
+    else
+      addCACerts(c.context, ca);
   } else {
     c.context.addRootCerts();
   }
 
-  const { cert } = options;
   if (cert) {
-    if (ArrayIsArray(cert)) {
-      for (const val of cert) {
-        validateKeyOrCertOption('cert', val);
-        c.context.setCert(val);
-      }
-    } else {
-      validateKeyOrCertOption('cert', cert);
-      c.context.setCert(cert);
-    }
+    if (ArrayIsArray(cert))
+      setCerts(c.context, ...cert);
+    else
+      setCerts(c.context, cert);
   }
 
   // Set the key after the cert.
   // `ssl_set_pkey` returns `0` when the key does not match the cert, but
   // `ssl_set_cert` returns `1` and nullifies the key in the SSL structure
   // which leads to the crash later on.
-  const key = options.key;
-  const passphrase = options.passphrase;
   if (key) {
     if (ArrayIsArray(key)) {
       for (const val of key) {
         // eslint-disable-next-line eqeqeq
         const pem = (val != undefined && val.pem !== undefined ? val.pem : val);
-        validateKeyOrCertOption('key', pem);
-        c.context.setKey(pem, val.passphrase || passphrase);
+        setKey(c.context, pem, val.passphrase || passphrase);
       }
     } else {
-      validateKeyOrCertOption('key', key);
-      c.context.setKey(key, passphrase);
+      setKey(c.context, key, passphrase);
     }
   }
 
-  const sigalgs = options.sigalgs;
   if (sigalgs !== undefined) {
-    if (typeof sigalgs !== 'string') {
+    if (typeof sigalgs !== 'string')
       throw new ERR_INVALID_ARG_TYPE('options.sigalgs', 'string', sigalgs);
-    }
 
-    if (sigalgs === '') {
+    if (sigalgs === '')
       throw new ERR_INVALID_ARG_VALUE('options.sigalgs', sigalgs);
-    }
```

Comment:
Non-blocking, but these seem like unrelated style changes.

Reply:
As this is non-blocking, I prefer to keep this as is